test/test_puma_server.rb

```diff
@@ -383,6 +383,20 @@ def test_early_hints_is_off_by_default
     assert_equal expected_data, data
   end
 
+  def test_request_payload_too_large
+    server_run(http_content_length_limit: 10)
+
+    sock = send_http "POST / HTTP/1.1\r\nHost: test.com\r\nContent-Type: text/plain\r\nContent-Length: 19\r\n\r\n"
+    sleep 0.5
```

Comment:
Are these sleeps necessary? 🤔

Reply:
actually, not in this case, i saw some other tests were doing it for replicating a timeout, but its not necessary here. Will update.